project2/os.c

```diff
@@ -12,12 +13,18 @@ void OS_Abort(ABORT_CODE error) {
     // Disable interrupts
     OS_DI();
     Kernel_Abort();
+    UART_Init0(38400);
 
     // Blink the built-in LED in accordance with the error code
     BIT_SET(DDRB, 7); // Set PB7 as output
     BIT_CLR(PORTB, 7);
     uint8_t ctr;
+
+    char buffer[20];
+    sprintf(buffer, "OS Abort. Error code: %d\n", error);
```

Comment:
Buffer too short for string (in js: `"OS Abort. Error code: XX\n".length === 25`, it's 24 without the newline)

Reply:
🤦‍♂️